board/drivers/can.h

```diff
@@ -57,8 +57,8 @@ int can_overflow_cnt = 0;
 
 // ********************* interrupt safe queue *********************
 
-int can_pop(can_ring *q, CAN_FIFOMailBox_TypeDef *elem) {
-  int ret = 0;
+bool can_pop(can_ring *q, CAN_FIFOMailBox_TypeDef *elem) {
+  bool ret = 0;
```

Comment:
false?

Reply:
zero automatically cats into false and non-zero into true and misra does not complain. It will be quite a bit of work to be consistent with bool assignments everywhere (I started it, then I gave up :) ).
I can change it for this function though.